dashboard/src/components/BaseCharts/types.ts

```
@@ -0,0 +1,75 @@
+  /**
+   * echarts opts config, default is `{}`
+   */
+  readonly opts?: Opts;
+  /**
+   * when after chart reander, do the callback with echarts instance
+   */
+  readonly onChartReady?: (instance: EChartsInstance) => void;
+  /**
+   * bind events, default is `{}`
+   */
+  readonly onEvents?: Record<string, { query?: string | Record<string, any>; function: Function }>;
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)